File: src/hes/sudoku/Group.java
package hes.sudoku;

import java.util.*;

public class Group {

    private final String name;
    private final Set<Cell> cells = new LinkedHashSet<>(9);

    public Group(String name) {
        this.name = name;
    }

    public Set<Cell> getCells() {
        return Collections.unmodifiableSet(cells);
    }

    public void addCell(Cell cell) {
        cells.add(cell);
        cell.add(this);
    }

    public void eliminate(Integer number) {
        cells.forEach(c -> c.eliminate(number));
    }

    public int candidates() {
        int candidates = 9;
        for (Cell cell : cells) {
            if (cell.isKnown()) {
                candidates--;
            }
        }

        return candidates;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: src/main/java/hes/sudoku/kt/Group.kt
package hes.sudoku.kt

class Group(val name: String) {

    private val _cells = mutableSetOf<Cell>()

    val cells: Set<Cell> = _cells

    fun addCell(cell: Cell) {
        _cells.add(cell)
        cell.add(this)
    }

    fun eliminate(number: Int?) {
        _cells.forEach { it.eliminate(number) }
    }

    override fun toString(): String {
        return name
    }
}